File: src/main/java/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtFunctionMixin.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins;

import com.demonwav.mcdev.platform.mcp.at.gen.psi.AtArgument;
import com.demonwav.mcdev.platform.mcp.at.gen.psi.AtFuncName;
import com.demonwav.mcdev.platform.mcp.at.gen.psi.AtReturnValue;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface AtFunctionMixin extends PsiElement {

    @NotNull
    List<AtArgument> getArgumentList();

    @NotNull
    AtFuncName getFuncName();

    @NotNull
    AtReturnValue getReturnValue();

    void setArgumentList(@NotNull String argumentList);

    void setReturnValue(@NotNull String returnValue);

    void setFunction(@NotNull String function);
}

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtFunctionMixin.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins

import com.demonwav.mcdev.platform.mcp.at.gen.psi.AtArgument
import com.demonwav.mcdev.platform.mcp.at.gen.psi.AtFuncName
import com.demonwav.mcdev.platform.mcp.at.gen.psi.AtReturnValue
import com.demonwav.mcdev.platform.mcp.at.psi.AtElement

interface AtFunctionMixin : AtElement {

    val argumentList: List<AtArgument>
    val funcName: AtFuncName
    val returnValue: AtReturnValue

    fun setArgumentList(argumentList: String)
    fun setReturnValue(returnValue: String)
    fun setFunction(function: String)
}
